File: archive/unique-number-of-occurrences.js
const uniqueOccurrences = function(arr) {
  let mem = {}
  for (let i = 0; i < arr.length; i++) {
    if (mem[arr[i]] === undefined) {
      mem[arr[i]] = 1
    } else {
      mem[arr[i]]++
    }
  }
  let mem2 = []
  Object.keys(mem).forEach(k => {
    mem2.push(mem[k])
  })
  let set = new Set(mem2)

  return (set.size === mem2.length)
};

console.log(uniqueOccurrences([1,2,2,1,1,3]))

File: go-archive/unique-number-of-occurrences.go
func uniqueOccurrences(arr []int) bool {
    mem := map[int]int{}

    for i:=0; i < len(arr); i++ {
        if _, ishas := mem[arr[i]]; ishas {
            mem[arr[i]]++
        } else {
            mem[arr[i]] = 1
        }
    }

    c:=map[int]bool{}

    for _, v := range mem {
        if _, ishas := c[v]; ishas {
            return false
        } else {
            c[v] = true
        }
    }

    return true
}